torch/csrc/jit/codegen/cuda/lower_insert_syncs.cpp

```diff
@@ -700,7 +700,13 @@ class ReadAfterWriteSyncs : public kir::ExprMutator {
         //  to insert the block sync. see FusionRAWSyncInsertionPlace4.
         std::unordered_set<Expr*> smem_writes;
         for (auto it : smem) {
-          smem_writes.insert(it.second);
+          // No need to handle writes that does not require block sync.
+          if (!GpuLower::current()
```

Comment:
nit: I think it's more intuitive to do: `GpuLower::.....needsRawSync(it.first->as<TensorView>()).any()`

Reply:
Makes sense. Further updated to `needsRawSync(...).hasTID`, since other bit fields don't seem to be relevant to shared mem and may introduce noise.